src/main/java/com/pgs/booking/controller/UserController.java

```diff
@@ -0,0 +1,43 @@
+package com.pgs.booking.controller;
+
+import com.pgs.booking.model.dto.CreateUserDto;
+import com.pgs.booking.model.dto.UserDto;
+import com.pgs.booking.service.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("api/users")
+public class UserController {
+
+    private final UserService userService;
+
+    @PostMapping
+    public UserDto addUser(@Valid @RequestBody CreateUserDto createUserDto) {
+        return userService.addUser(createUserDto);
+    }
+
+    @GetMapping({"/{id}"})
+    public UserDto getSingleUser(@Valid @PathVariable Long id) {
+        return userService.getSingleUser(id);
+    }
+
+    @PostMapping("/{id}/activate")
+    public UserDto activateUser(@PathVariable Long id) {
+        return userService.activateUser(id);
+    }
+
+    @PostMapping("/{id}/deactivate")
+    public UserDto deactivateUser(@PathVariable Long id) {
+        return userService.deactivateUser(id);
+    }
+
+    @PostMapping("/{id}/setRoles")
+    public UserDto setUserRoles(@PathVariable Long id, @RequestBody List<String> roleDtoList) {
+        return userService.setUserRoles(id, roleDtoList);
+    }
```

Comment:
Usually editing an entity is under PUT method
```suggestion
    @PutMapping("/{id}/activate")
    public UserDto activateUser(@PathVariable Long id) {
        return userService.activateUser(id);
    }

    @PutMapping("/{id}/deactivate")
    public UserDto deactivateUser(@PathVariable Long id) {
        return userService.deactivateUser(id);
    }

    @PutMapping("/{id}/setRoles")
    public UserDto setUserRoles(@PathVariable Long id, @RequestBody List<String> roleDtoList) {
        return userService.setUserRoles(id, roleDtoList);
    }
```